Firestore/core/src/firebase/firestore/local/serializer.cc

```diff
@@ -44,8 +47,7 @@ void Serializer::EncodeMaybeDocument(
       writer->WriteTag(
           {PB_WT_STRING, firestore_client_MaybeDocument_document_tag});
       writer->WriteNestedMessage([&](Writer* writer) {
-        const model::Document& doc =
-            static_cast<const model::Document&>(maybe_doc);
+        auto doc = static_cast<const model::Document&>(maybe_doc);
```

Comment:
This should be `const auto&`. Plain `auto` will make a copy here.

The rules for `auto` deduction are similar to template argument deduction:
```cpp
template <typename T>
void foo(T t);
template <typename T>
void bar(const T& t);
const std::string& str_ref = some_global_string;
foo(str_ref); // copies -- T deduces to `std::string`, so the argument is `std::string`
auto foo2 = str_ref; // copies -- auto deduces to `std::string`
bar(str_ref); // no copy -- T deduces to `std::string`, so the argument is `const std::string&`
const auto& bar2 = str_ref; // no copy
```
Sorry about that. If you feel like specifying full type makes it easier to avoid this issue, feel free to revert to how it was before.

Reply:
Nah, this is just me being dumb and not paying enough attention to what I'm doing. I'll keep the auto here (but will add the &). Thanks for picking up on this.

Reply:
Now deleted altogether.